gh-98298: [Enum] document ReprEnum, global_enum, and show_flag_values (#98455)
python/cpython: Doc/library/enum.rst

      the bitwise operators without losing their :class:`IntFlag` membership.
      :class:`IntFlag` members are also subclasses of :class:`int`. (`Notes`_)

   :class:`ReprEnum`

      Used by ``IntEnum``, ``StrEnum``, and ``IntFlag`` to keep the ``str()`` of

>       the mixed-in type.

Reviewer: ```suggestion
      Used by :class:`IntEnum`, :class:`StrEnum`, and :class:`IntFlag`
      to keep the :class:`str() <str>` of the mixed-in type.
```

Add Sphinx cross-references

      Do not make ``obj`` a member.  Can be used as a decorator.

   :func:`global_enum`

      Modify the ``str()`` and ``repr()`` of an enum to show its module instead of
      its class -- should only be used if the enum members will be exported to the

>       global namespace.

Reviewer: ```suggestion
      Modify the :class:`str() <str>` and :func:`repr` of an enum
      to show its members as belonging to the module instead of its class.
      Should only be used if the enum members will be exported to the
      module global namespace.
```

* Clarified more specifically what it does
* Prepended "module" to global namespace to make the scope more clear; I at least found it confusing at first
* Split it into two sentences to read a little better
* Added Sphinx syntax

      better support the *replacement of existing constants* use-case.
      :meth:`__format__` was already :func:`int.__format__` for that same reason.

.. class:: ReprEnum

>    *ReprEum* uses the *repr()* of *Enum*, but the *str()* of the mixed-in data type:

Reviewer: ```suggestion
   :class:`!ReprEum` uses the :meth:`repr() <Enum.__repr__>` of :class:`Enum`,
   but the :class:`str() <str>` of the mixed-in data type:
```

Add standard Sphinx syntax—FYI, `!` keeps the same consistent styling of a class (or other name) but suppresses the (undesired) reference link

>       * :meth:`int.__str__` for *IntEnum* and *IntFlag*
>       * :meth:`str.__str__` for *StrEnum*

Reviewer: ```suggestion
      * :meth:`!int.__str__` for :class:`IntEnum` and :class:`IntFlag`
      * :meth:`!str.__str__` for :class:`StrEnum`
```

Sphinx syntax

>    Inherit from *ReprEnum* when mixing in a data type when the *str()* output should
>    not be changed to the *Enum*-default *str()*.

Reviewer: ```suggestion
   Inherit from :class:`!ReprEnum` when mixing in a data type
   when the :class:`str() <str>` output should not be changed
   to the :class:`Enum`-default :meth:`str() <Enum.__str__>`.
```

Sphinxify using standard syntax

      better support the *replacement of existing constants* use-case.
      :meth:`__format__` was already :func:`int.__format__` for that same reason.

.. class:: ReprEnum

   :class:`!ReprEum` uses the :meth:`repr() <Enum.__repr__>` of :class:`Enum`,
   but the :class:`str() <str>` of the mixed-in data type:

>       * :meth:`int.__str__` for :class:`IntEnum` and :class:`IntFlag`
>       * :meth:`str.__str__` for :class:`StrEnum`

Reviewer: ```suggestion
      * :meth:`!int.__str__` for :class:`IntEnum` and :class:`IntFlag`
      * :meth:`!str.__str__` for :class:`StrEnum`
```

Seems like this one might have been missed when my suggestions were apparently being re-implemented manually

>    Inherit from :class:`!ReprEnum` when mixing in a data type when the
>    :class:`str() <str>` output should not be changed to the

Reviewer: ```suggestion
   Inherit from :class:`!ReprEnum` to keep the :class:`str() <str> / :func:`format`
   of the mixed-in data type instead of using the
```

After giving this a closer look, I found my original confusing and think it reads much clearer this way (and avoids a double "when")

   .. versionadded:: 3.11

.. decorator:: global_enum

>    A decorator to change the *str()* and *repr()* of an enum to show the module
>    of the enum instead of its class.  Should only be used when the enum members
>    are exported to the global namespace (see *re.RegexFlag* for an example).

Reviewer: ```suggestion
   A decorator to change the :class:`str() <str>` and :func:`repr` of an enum
   to show its members as belonging to the module instead of its class.
   Should only be used when the enum members are exported
   to the module global namespace (see class:`re.RegexFlag` for an example).
```

* Tweaks to wording to clarify as previous
* Sphinxify

   .. versionadded:: 3.11

> .. function:: show_flag_values

Reviewer: ```suggestion
.. function:: show_flag_values(value)
```

Add function signature

>    Return a list of all power-of-two integers contained in a flag.

Reviewer: ```suggestion
   Return a list of all power-of-two integers contained in a flag *value*.
```

Add a reference to the parameter.

   .. versionadded:: 3.11

.. decorator:: global_enum

   A decorator to change the :class:`str() <str>` and :func:`repr` of an enum
   to show its members as belonging to the module instead of its class.
   Should only be used when the enum members are exported

>    to the module global namespace (see class:`re.RegexFlag` for an example).

Reviewer: ```suggestion
   to the module global namespace (see :class:`re.RegexFlag` for an example).
```

Silly typo here, as I feared—my fault.
